src/query_param.c

```diff
@@ -159,7 +159,15 @@ void QueryParam_InitParams(QueryParam *p, size_t num) {
   memset(p->params, 0, sizeof(*p->params) * num);
 }
 
-int QueryParam_Resolve(Param *param, dict *params, QueryError *status) {
+int checkNumericValueValid(const char *val, unsigned int dialectVersion) {
+    if (dialectVersion >= 2 && (!*val)) {
+      return 0;
+    }
+
+  return 1;
+}
```

Comment:
This function is the center of this PR
1. Let's keep the indentation in the file coherent (double-space per indentation)
2. Comment on what we are validating, and why it's bound to the dialect of the query
3. Since it's the only validation here, consider having it inlined:
```suggestion
static inline bool checkNumericValueValid(const char *val, unsigned int dialectVersion) {
  return (dialectVersion < 2 || *val);
}
```